include/oneapi/dpl/pstl/hetero/dpcpp/utils_ranges_sycl.h

```diff
@@ -206,6 +207,16 @@ struct is_passed_directly<Iter, ::std::enable_if_t<Iter::is_passed_directly::val
 {
 };
 
+//support std::vector::iterator with usm host / shared allocator as passed directly
+template <typename Iter>
+struct is_passed_directly<
+    Iter, std::enable_if_t<(std::is_same_v<Iter, oneapi::dpl::__internal::__usm_shared_alloc_vec_iter<Iter>> ||
```

Comment:
Should we apply here `std::decay_t` for `Iter` template param?

Reply:
Same as above, this is merged code from #1438.  The duplicated conflict code has been fixed via rebase.  Any changes here will need a new PR.